biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFileRenamer.java

```diff
@@ -0,0 +1,588 @@
+package aQute.bnd.classfile;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import aQute.bnd.classfile.ConstantPool.AttributeVisitor;
+import aQute.bnd.classfile.ConstantPool.ClassInfo;
+import aQute.bnd.classfile.ConstantPool.EntryVisitor;
+import aQute.bnd.classfile.ConstantPool.MethodTypeInfo;
+import aQute.bnd.classfile.ConstantPool.NameAndTypeInfo;
+import aQute.bnd.classfile.InnerClassesAttribute.InnerClass;
+import aQute.bnd.classfile.LocalVariableTableAttribute.LocalVariable;
+import aQute.bnd.classfile.LocalVariableTypeTableAttribute.LocalVariableType;
+import aQute.bnd.classfile.builder.ClassFileBuilder;
+import aQute.bnd.classfile.builder.MutableConstantPool;
+import aQute.bnd.classfile.preview.RecordAttribute;
+import aQute.bnd.classfile.preview.RecordAttribute.RecordComponent;
+import aQute.lib.strings.Strings;
+
+class ClassFileRenamer {
+	final Function<String, String>	mapper;
+	final Map<Integer, String>		renamed	= new HashMap<>();
+	final ClassFile					classFile;
+	final ClassFileBuilder			builder;
+	final SignatureRenamer			sp;
+	boolean							changed	= false;
+
+	static Optional<ClassFile> rename(ClassFile cf, Function<String, String> mapper) {
+		ClassFileRenamer classFileRenamer = new ClassFileRenamer(cf, mapper);
+		return classFileRenamer.rename();
+	}
+
+	static class SignatureParser {
```

Comment:
Why was the signature package not used? We now have to signature parsers and bug sources?